frontend/src/components/single-blog/single.blog.jsx

```diff
@@ -1,58 +1,77 @@
-import React from "react";
-
 import { SectionHeader, BlogArticles, AboutDetails, Container } from "shared-components";
+import BlogsService from "pages/api/blogsService";
+import { useState, useEffect } from "react";
+
+var loaded = false;
+var currentBlogId = 0;
+
+var buffer = {
+  id: 0,
+  sections: [],
+  categories: [],
+  author: {
+    id: 0,
+    created: "",
+    modified: "",
+    first_name: "",
+    last_name: "",
+    description: "",
+    facebook_url: "",
+    image: null,
+    instagram_url: "",
+    blog: 0,
+  },
+  created: "",
+  modified: "",
+  title: "",
+  image: "",
+  description: "",
+};
+
+export const SingleBlog = ({ blogId }) => {
+  const [blog, setBlog] = useState(buffer);
+
+  async function getBlog(blogId) {
+    const response = await BlogsService.getBlogById(blogId);
+    setBlog(response.data);
+  }
+  if (blogId != undefined && (!loaded || currentBlogId != blogId)) {
+    currentBlogId = blogId;
+    getBlog(blogId);
```

Comment:
Moze pojasnjenje sto je ovde rekurzivna funkcija. Cini mi se da ono sto zelis da postignes u ovom if-u mozes resiti preko useEffect hooka

Reply:
Kada se prvi put stranica ucita, prikazuje podatke iz `buffer` promenljive i `loaded = false`. Ovaj if proverava da li je API pozvan i da li su podaci ucitani, ako nisu onda se pozove `getBlog(blogId)`. Nebi trebalo da je funkcija rekurzivna.

Reply:
Ovo bi svakako trebalo staviti u `useEffect`. 
Takodje, `loaded` i `currentBlogId` bi trebalo da budu unutar komponente i da se inicijalizuju i mutiraju (update-uju) pomocu `useState`.
Treca stvar, bool varijable treba da imaju prefiks kao sto su: `is`, `has`, `should` itd. a u tvom slucaju bi najbolje bilo da ti se `loaded` zove `hasLoaded` ili `isLoading`

Reply:
Razumem, preimenovacu varijable, ali nisam siguran kako se koristi useEffect u ovom slucaju. Da li postoji neki primer u codebase-u ili da trazim na netu neko resenje?

Reply:
Pogledaj `blog-filters/blog.filter.jsx`. Mozes iskoristiti taj primer.